### PR TITLE
Artifactory Collector - artifact based

### PR DESCRIPTION
**Pull request by nireeshT**
This PR includes changes to 
1) Changes in Artifactory collector to branch out logic to get artifacts at the level of repository and keeping current functionality is intact. 
2) Restructuring properties to better modularize the server settings(usernames, apikeys, repos and paterns) and enable the flag to set mode (possible values - REPO_BASED, ARTIFACT_BASED).Refer readme doc for properties file changes. 
3) Added more details in binary artifact. (createdBy, createdTimeStamp, actual_sha, modifiedBy, modifiedTimestamp)

### CODE REVIEW

File: collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/DefaultArtifactoryClient.java [hunk 1]
```diff
@@ -29,132 +26,220 @@
...
+
+					artifactPatterns.add(p);
+				} catch (PatternSyntaxException e) {
+					LOGGER.error("Invalid pattern: " + e.getMessage());
+					throw e;
+				}
+			}
+		}
+
+		if (artifactPatterns.isEmpty()) {
```
Comment (rvema) [inline collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/DefaultArtifactoryClient.java]: Can you ensure this is null safe?

Comment (nireeshT) [reply collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/DefaultArtifactoryClient.java]: @rvema  yes, it is null safe

File: collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/DefaultArtifactoryClient.java [hunk 2]
```diff
@@ -29,132 +26,220 @@
...
+						artifactPath = artifactPath.substring(0, artifactPath.length() - 1);
+					}
+
+					// create artifact_items (collector_item)
+					ArtifactItem artifactItem = new ArtifactItem();
+					artifactItem.setInstanceUrl(instanceUrl);
+					artifactItem.setRepoName(repo);
+					artifactItem.setArtifactName(artName);
+					artifactItem.setPath(artPath);
+					artifactItem.setDescription(artName);
```
Comment (nireeshT) [inline collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/DefaultArtifactoryClient.java]: @nireeshT  extract code to artifactItem method

File: collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/ArtifactoryCollectorTask.java [hunk 3]
```diff
@@ -1,150 +1,207 @@
...
+	}
+
+	@Override
+	public String getCron() {
+		return artifactorySettings.getCron();
+	}
+
+	@Override
+	public void collect(ArtifactoryCollector collector) {
+		if (artifactorySettings.getMode().equals(ArtifactoryCollectionMode.REPO_BASED)) {
```
Comment (nireeshT) [inline collectors/artifact/artifactory/src/main/java/com/capitalone/dashboard/collector/ArtifactoryCollectorTask.java]: @nireeshT  switch case

**Pull request merged**